BlazarService/src/main/java/com/hubspot/blazar/listener/SlackDmNotificationVisitor.java

```diff
@@ -0,0 +1,91 @@
+package com.hubspot.blazar.listener;
+
+
+import static com.hubspot.blazar.base.RepositoryBuild.State.FAILED;
+import static com.hubspot.blazar.base.RepositoryBuild.State.UNSTABLE;
+
+import java.util.Collections;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Sets;
+import com.google.inject.Inject;
+import com.hubspot.blazar.base.BuildTrigger;
+import com.hubspot.blazar.base.RepositoryBuild;
+import com.hubspot.blazar.base.visitor.RepositoryBuildVisitor;
+import com.hubspot.blazar.config.BlazarConfiguration;
+import com.hubspot.blazar.config.BlazarSlackConfiguration;
+import com.hubspot.blazar.util.SlackClientWrapper;
+import com.hubspot.blazar.util.SlackMessageBuildingUtils;
+
+/**
+ * Direct messages slack users about the state of builds caused by their pushes
+ */
+public class SlackDmNotificationVisitor implements RepositoryBuildVisitor {
+  // Cancelled requires a user action - likely the user who would be slacked anyway so it is not worth messaging about
+  private static final Set<RepositoryBuild.State> SLACK_WORTHY_FAILING_STATES = ImmutableSet.of(FAILED, UNSTABLE);
+  private static final Logger LOG = LoggerFactory.getLogger(SlackDmNotificationVisitor.class);
+  private final BlazarSlackConfiguration blazarSlackConfig;
+
+  private final SlackMessageBuildingUtils slackMessageBuildingUtils;
+  private final SlackClientWrapper slackClientWrapper;
+
+  @Inject
+  public SlackDmNotificationVisitor(BlazarConfiguration blazarConfiguration,
+                                    SlackMessageBuildingUtils slackMessageBuildingUtils,
+                                    SlackClientWrapper slackClientWrapper) {
+    this.slackMessageBuildingUtils = slackMessageBuildingUtils;
+    this.slackClientWrapper = slackClientWrapper;
+    this.blazarSlackConfig = blazarConfiguration.getSlackConfiguration().get();
+  }
+
+  @Override
+  public void visit(RepositoryBuild build) throws Exception {
+    Set<String> userEmailsToSendMessagesTo = getOptedInUserEmailsForBuild(build);
+    boolean shouldSendMessage = shouldSendMessage(build, userEmailsToSendMessagesTo);
+    if (shouldSendMessage) {
+      for (String email : userEmailsToSendMessagesTo) {
+        slackClientWrapper.sendMessageToUserByEmail(email, "", slackMessageBuildingUtils.buildSlackAttachment(build));
```

Comment:
slackClientWrapper.sendMessageToUserByEmail is a bit confusing (sounds like sending an email through slack). The method name could just be `sendMessage` (overloaded) like you have named the private methods (name the private methods `sendMessageWithoutRetries` or something like that)